src/sentry/coreapi.py

```diff
@@ -407,6 +407,11 @@ def should_filter(self, project, data, ip_address=None):
         if error_message and not is_valid_error_message(project, error_message):
             return (True, FilterStatKeys.ERROR_MESSAGE)
 
+        for exception_interface in data.get('sentry.interfaces.Exception', {}).get('values', []):
+            message = u': '.join(filter(None, map(exception_interface.get, ['type', 'value'])))
```

Comment:
This seems like a reimplementation of `Exception.to_python(data).to_string()` so maybe you should use that? Although I can understand not wanting to do that for performance reasons.

Another pattern that may or may not be useful is something I started doing in https://github.com/getsentry/sentry/pull/6417/files#diff-da365f8d913a2a9c688f1d2b2e540f2eR488 where I delegate the filtering decision to the interface itself.

Reply:
I actually hadn't considered this since I didn't think we had actually went through any significant validation at this point and didn't want to have to guard against the `InterfaceValidationError`. (I forgot that actually happens as part of `LazyData` now and [doesn't actually result in the interface being converted to an `Interface` object, just validated as one](https://github.com/getsentry/sentry/blob/438866759c5a2988be13331d0f862655fc15ad49/src/sentry/event_manager.py#L371-L372).)

[`to_string` does take an `event` parameter](https://github.com/getsentry/sentry/blob/438866759c5a2988be13331d0f862655fc15ad49/src/sentry/interfaces/exception.py#L287) though, which doesn't exist at this point, and it feels wrong to pass `None` in my opinion. It also [includes the stacktrace](https://github.com/getsentry/sentry/blob/438866759c5a2988be13331d0f862655fc15ad49/src/sentry/interfaces/exception.py#L294-L298) which I don't think we want to include as part of the pattern matching input here.

Looking at this, the implementations don't really match either (I'm not sure if this is a good or bad thing):

```python
In [3]: Exception.to_python({'type': None, 'value': 'value'}).to_string(event=None)
Out[3]: u'None: value'
In [4]: Exception.to_python({'type': 'value', 'value': None}).to_string(event=None)
Out[4]: u'value: None'
```

Reply:
I do think that delegating filtering to interfaces where possible is a good idea, though. I think if we were able to do that without having to double validate the interface that'd be the ideal longer-term solution here, but I'm hesitant to commit to that scope of a change now.

Reply:
yeah, perhaps if we kept the validated interface objects around for longer rather than immediately serializing them again, we could do something different here, but I'm happy for you to leave it as is